frontend_server_client/lib/src/frontend_server_client.dart

```
@@ -40,6 +40,15 @@ class FrontendServerClient {
   /// The [outputDillPath] determines where the primary output should be, and
   /// some targets may output additional files based on that file name (by
   /// adding file extensions for instance).
+  ///
+  /// When the [frontendServerPath] argument is provided, the frontend server
+  /// will be started from the specified file. The specified file can either be
+  /// a Dart source file or an AppJIT snapshot.
+  ///
+  /// When the [frontendServerPath] argument is provided, setting [debug] to
```

Generally I would probably prefer for it to throw an error rather than just ignore the option.

Technically we would have to do a breaking change though, but I think it is unlikely many people will see this comment and would be confused that trying to just set `debug: true` doesn't do anything.

---

I've updated the PR to make `FrontendServerClient.start` throw an error when this case is hit. Since it's a breaking change, I've updated the version to `4.0.0-wip`, which means that `3.3.0` should be published before this PR can land.